src/main/java/net/neoforged/neoforge/client/settings/KeyMappingLookup.java

```diff
@@ -53,7 +54,7 @@ private List<KeyMapping> findKeybinds(InputConstants.Key keyCode, KeyModifier mo
     public void put(InputConstants.Key keyCode, KeyMapping keyBinding) {
         KeyModifier keyModifier = keyBinding.getKeyModifier();
         Map<InputConstants.Key, Collection<KeyMapping>> bindingsMap = map.get(keyModifier);
-        Collection<KeyMapping> bindingsForKey = bindingsMap.computeIfAbsent(keyCode, k -> new ArrayList<>());
+        Collection<KeyMapping> bindingsForKey = bindingsMap.computeIfAbsent(keyCode, k -> Sets.newConcurrentHashSet());
```

Comment:
Changing this from a list to a hash set may change the behaviour of multiple keybinds on a single key, including making their order random since `KeyMapping` doesn't override `equals()` and `hashCode()`. The safest option would be to use a `CopyOnWriteArrayList` instead.

Reply:
My gut instinct thinks that should be fine, but it's also has a greater overhead, especially since remove can be called often depending on what the user does on the keybinds screen. Based on some previous testing, I can't imagine it taking more than ms or two, if that, but I'll double check with a stress test.

Reply:
So, after a quick stress test of two mods registering about a 1,000 key mappings (this is just to increase potentially concurrency collisions), on average its usually around ~5ms, though its variance is wider depending on how many times a conflict occurs on an individual key. It's only slightly less than using a `synchronized` block for the array list with the concurrent map. Compared to that, the hash set is typically around ~2ms.

I also performed another test with ten mods and 100 key mappings for a more "realistic" scenario of the COW list on number of entries ranges between 0-2ms, which seems reasonable enough